app/queries/get_activity_log_events.rb

```diff
@@ -35,41 +35,46 @@ class GetActivityLogEvents
   def self.call(application_choices:, since: nil)
     since ||= application_choices.includes(:application_form).minimum('application_forms.created_at')
 
-    application_choices_join_sql = <<~COMBINE_AUDITS_WITH_APPLICATION_CHOICES_SCOPE_AND_FILTER.squish
-      INNER JOIN (#{application_choices.to_sql}) ac
-        ON (
-          auditable_type = 'ApplicationChoice'
-          AND auditable_id = ac.id
-          AND action = 'update'
-          AND ( #{application_choice_audits_filter_sql} )
-        ) OR (
-          associated_type = 'ApplicationChoice'
-          AND associated_id = ac.id
-          AND NOT auditable_type = 'OfferCondition'
-          AND NOT auditable_type = 'ApplicationExperience'
-          AND NOT auditable_type = 'ApplicationWorkHistoryBreak'
-        ) OR (
-          auditable_type = 'ApplicationForm'
-          AND auditable_id = ac.application_form_id
-          AND action = 'update'
-          AND ( #{application_form_audits_filter_sql} )
-          AND EXISTS (
-            SELECT 1
-            WHERE ARRAY[#{DATABASE_CHANGE_KEYS}] @> (
-              SELECT ARRAY(SELECT jsonb_object_keys(a.audited_changes)
-              FROM audits a
-              WHERE a.id = audits.id
+    application_form_changes_exist = <<~SQL
+      EXISTS (
+              SELECT 1
+              WHERE ARRAY[#{DATABASE_CHANGE_KEYS}] @> (
+                SELECT ARRAY(SELECT jsonb_object_keys(a.audited_changes)
+                FROM audits a
+                WHERE a.id = audits.id
+                )
               )
             )
-          )
-        )
-    COMBINE_AUDITS_WITH_APPLICATION_CHOICES_SCOPE_AND_FILTER
-
-    Audited::Audit.select('audits.id audit_id, audits.*, ac.id application_choice_id')
-                  .includes(INCLUDES)
-                  .joins(application_choices_join_sql)
-                  .where('audits.created_at >= ?', since)
-                  .order('audits.created_at DESC')
+    SQL
+
+    where_auditable_is_application_form = Audited::Audit
+      .where(auditable_id: application_choices.pluck(:application_form_id).uniq, auditable_type: 'ApplicationForm', action: :update)
```

Comment:
```suggestion
      .where(auditable_id: application_choices.select(:application_form_id).distinct, auditable_type: 'ApplicationForm', action: :update)
```
This was suggested yesterday to avoid 2 queries being executed

Reply:
Done